core/src/Kokkos_CopyViews.hpp

```diff
@@ -3418,6 +3418,7 @@ create_mirror_view_and_copy(
         Impl::MirrorViewType<Space, T, P...>::is_same_memspace>::type* =
         nullptr) {
   (void)name;
+  typename Space::execution_space().fence();
```

Comment:
cc @crtrott 
if we agree we want to fence here, there will be other overloads that need fences.

Reply:
Also this belongs to another separate pull request.